### DISCUSSION TITLE
Problems with `cobbler import` when `--available-as` is specified (problems with import_rsync_whitelist)

### DISCUSSION THREAD

**Discussion by cinnion** [cobbler/cobbler#3841] [Q&A]
Guten Tag!

Mainly posting this to see how many folks use `cobbler import` with the `--available-as` option. It seems that the trailing `- *` rule in `import_rsync_whitelist` which is only used when this option is specified is causing important sub-trees to be excluded, such as the `BaseOS` directory which is the signature for many of the later RHEL based distros. This is because, per the "Include/Exclude Pattern Rules" section of the [rsync manpage](https://linux.die.net/man/1/rsync), it states:

> If a pattern excludes a particular parent directory, it can render a deeper include pattern ineffectual because rsync did not descend through that excluded section of the hierarchy. This is particularly important when using a trailing "*" rule.

The problem is, this means that we would have to explicitly add include rules for each and every directory down to the packages we want, as opposed to rsyncing down gigabytes worth of data.  I can certainly add those rules for Rocky 9 (which will likely be valid for more of the RHEL based distros), but this leaves other distros untouched, with the `--available-as` option potentially useless.

Any thoughts??

**Comment by SchoolGuy** [comment]
Today in Europe the day is almost over and I certainly am tired, I'll get back to you on Tuesday when I was able to reproduce it.

My understanding from your question is that you think that you found a bug that certain folder structures are missing when using the available-as flag. Is that a correct description? If yes what would be the desired behaviour that you expect?

**Reply by cinnion** [reply]
Thanks Enno. That is correct.  When I import Rocky9, the directory BaseOS is not rsync'ed into `/var/www/cobbler/distro_mirror/distro-name/` when the flag is given, so cobbler reports that no signatures are found, and no distro and associated profile are created. Below are the diffs which fix it apparently for importing RL9 both from the DVD and a local copy of the RL9 mirror pulled down via rsync when I specify that flag. With these diffs, the import succeeds and I get both the distro and profile Handling the creation of a repo (which I vaguely remember seeing someone mention at some point) is trivial, though perhaps with the `available-as` flag if we give it, maybe we also have enough to at least create a minimally defined repo as well if we needed/wanted it. Have not thought that through completely yet... all the repos I have defined have been locally created add-on packages ages ago.

The diffs so far are...
```
diff --git a/import_rsync_whitelist b/import_rsync_whitelist
index e33d185..d2baaae 100644
--- a/import_rsync_whitelist
+++ b/import_rsync_whitelist
@@ -23,6 +23,12 @@
 + SL/
 + FermiPackages/
 + SITERPMS/
++ BaseOS/
++ BaseOS/*/
++ BaseOS/*/os/
++ BaseOS/*/os/images/
++ BaseOS/*/os/images/pxeboot/
++ BaseOS/*/os/images/pxeboot/*
 + images/
 + images/pxeboot/
 + images/pxeboot/*
```

Also wondering why we pull the kernel RPMs in in this case...we don't really need them, just the kernel and initrd.

Guten nacht Enno!
